src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -22,28 +24,45 @@
 using Microsoft.PowerFx.Core.Utils;
 using Microsoft.PowerFx.Syntax;
 using Microsoft.PowerFx.Types;
-using static Microsoft.PowerFx.Core.Localization.TexlStrings;
+using static Microsoft.PowerFx.Core.Localization.TexlStrings;
+using CallNode = Microsoft.PowerFx.Syntax.CallNode;
 
 namespace Microsoft.PowerFx.Core.Functions
-{
+{
     /// <summary>
     /// Represents a user defined function (UDF) - which is created from parsing other Power Fx.     
     /// This includings the binding (and hence IR for evaluation) - 
     /// This is conceptually immutable after initialization - if the body or signature changes, you need to create a new instance.
     /// </summary>
-    internal class UserDefinedFunction : TexlFunction
+    internal class UserDefinedFunction : TexlFunction, IExternalPageableSymbol, IExternalDelegatableSymbol
     {
         private readonly bool _isImperative;
         private readonly IEnumerable<UDFArg> _args;
         private TexlBinding _binding;
 
-        public override bool IsAsync => _binding?.IsAsync(UdfBody) ?? false;
+        public override bool IsAsync => _binding?.IsAsync(UdfBody) ?? false;
+
+        public bool IsPageable => _binding?.IsPageable(_binding.Top) ?? false;
+
+        public bool IsDelegatable => _binding?.IsDelegatable(_binding.Top) ?? false;

```

Comment:
Can you add few binding tests with simple and deeply nested expressions using this mock datasource [https://github.com/microsoft/Power-Fx/blob/c24014662051e2fc6740e4c291f26210e1bfcf37/src/tests/Microsoft.PowerFx.Core.Tests.Shared/Helpers/TestTabularDataSource.cs#L293](https://github.com/microsoft/Power-Fx/blob/c24014662051e2fc6740e4c291f26210e1bfcf37/src/tests/Microsoft.PowerFx.Core.Tests.Shared/Helpers/TestTabularDataSource.cs#L293) ?